inbox/models/backends/imap.py

```diff
@@ -209,11 +209,19 @@ def update_labels(self, new_labels: List[str]) -> None:
     def namespace(self):
         return self.imapaccount.namespace
 
-    @property
-    def categories(self) -> Set[Category]:
-        categories = {label.category for label in self.labels}
-        categories.add(self.folder.category)
-        return categories
+    def get_categories(self, category_type: CategoryType) -> set[Category]:
+        """
+        Returns the categories this uid belongs to.
+
+        Args:
+            category_type: The category type for the account this uid belongs to.
+        """
+        if category_type == "label":
+            return {label.category for label in self.labels} | {self.folder.category}
+        elif category_type == "folder":
+            return {self.folder.category}
+        else:
+            assert_never(category_type)
```

Comment:
Here we were unconditionally loading labels, even if the account does not support labels (only Gmail supports them), this was just stupid and I parametrized this on category_type to avoid this.

Reply:
Additionally this would be a perfect place to use match and case but I need to update our lint tooling to support that syntax and there were too many things to change to make it work on this PR.